### ISSUE TITLE
SnakeYAML 2.1+ is not compatible with ElasticJob Lite

### ISSUE THREAD

**Issue by wm0018** [apache/shardingsphere-elasticjob#2264]
We upgraded from SnakeYAML 1.x to 2.x due to security vulnerabilities. However, this upgrade introduced some incompatibilities. Can we explore the possibility of making it compatible again?

**Comment by linghengqian** [comment]
- This is actually handled at https://github.com/apache/shardingsphere-elasticjob/pull/2216 , are you saying we should bump to SnakeYAML 2.2?

**Comment by wm0018** [comment]
> * This is actually handled at [Upgrade snakeyaml to 2.0 and springboot to 2.7.10 #2216](https://github.com/apache/shardingsphere-elasticjob/pull/2216) , are you saying we should bump to SnakeYAML 2.2?

yes，snakeyaml 2.x is not compatible

**Comment by linghengqian** [comment]
- In fact, the Spring Framework that conflicts with this does not advance to SnakeYAML 2.0 until 6.1.0-SNAPSHOT, where they ensure binary compatibility of SnakeYAML versions by overriding Classpath in unit tests.

- I will try to introduce this process in the unit tests of ElasticJob's Spring Boot Starter. Before I can do that, I need to migrate the project from Junit Vintage to Junit Jupiter.

**Comment by linghengqian** [comment]
- @zhfeng Hi, I noticed the #2216 for the first time introduced `org.yaml.snakeyaml.inspector.TrustedPrefixesTagInspector`.
- However, this class has been removed in SnakeYAML 2.1, and there is not much javadoc information. Refer to https://bitbucket.org/snakeyaml/snakeyaml/wiki/Changes and https://bitbucket.org/snakeyaml/snakeyaml/commits/39a4b874c70ddf8beb0b73c4092ea1c700cc66df .
- This causes ElasticJob to break binary compatibility on SnakeYAML 2.1+. Do we have to use this class?
